cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/PersistedMessageCallback.java

```diff
@@ -120,10 +112,7 @@ private void recordLastOffset() {
       if (lastOffset >= 0) {
         offsets.put(lastTopicPartition, lastOffset);
       }
-      if (messageCount.get() >= persistThreshold) {
```

Comment:
Yes, I think so too. Then we need to find out if persisting offsets is transactional or not.